src/dmd/dmodule.d

```diff
@@ -1035,6 +1054,20 @@ extern (C++) final class Module : Package
         {
             isHdrFile = true;
         }
+
+        /* If it has the extension ".c", it is a "C" file.
+         */
+        if (FileName.equalsExt(arg, c_ext))
```

Comment:
What about C header files? Code using external libraries might only have access to the header files, not the actual sources.

Reply:
One thing at a time.